Documentation/Communicational Patterns/2. Tooling/2.1. Tooling.md

```diff
@@ -0,0 +1,133 @@
+![document type: vision](https://themomax.github.io/resources/markdown-labels/document_type_vision.svg)
+
+# Tooling
+
+## Component
+
+A service consists of multiple endpoints. Apodini provides a DSL that allows to define the service's structure using `Component`s. The `Component`-stack is determined on startup and may not change during runtime. `Component`s that provide a `handler` instead of the recursive `content` property define endpoints. Both, input- and output-type of an endpoint are fixed at startup.
+
+## Handler
+
+Endpoints are implemented using `Handler`s. A `Handler` has an associated type `Response: ResponseEncodable`. Implementations of the `Handler` protocol are `struct`s which provide a function `handle() -> Action`. `Action` is an enum that can either be `nothing`, `send(Response)`, `end` or `final(Response)`.
+
+A default exists that transforms the result of `handle() -> Response` to `final(handle())`.
+
+### Lifetime
+
+In general a `Handler` is kept alive until it was evaluated to `Action.final(Response)` or `Action.end`. Some exporters may enforce further rules based on the communicational patterns that are used to represent the respective endpoint. E.g. a HTTP 1 based exporter must export an endpoint that could send multiple service-messages to the client after an initial request. As the request-response pattern is the only pattern HTTP 1 supports, the exporter downgrades the endpoint to the request-response pattern. In this scenario the exporter could destruct the `Handler` after the first `Action.send(Response)`, because it cannot send more messages anyways.
+
+### Dynamic Properties
+
+`Handler`s can use different types of `DynamicProperty` variables. Apodini provides multiple implementations of this protocol that provide different functionality. The implementations are described in the following chapters. Properties on `Handler`s that are a `DynamicProperty` are managed by the Apodini framework. The framework makes sure they are exclusive to one client-service connection. The developer does not have to take care of synchronization. Other properties should be used for static, globally shared content only or references to services that manage access and synchronization themselves.
+
+### Layers of Logic
```

Comment:
I liked the idea of a `Maybe` section for proposals and documentation that @Supereg and @nerdsupremacist had in that proposal: https://github.com/Apodini/Apodini/pull/41#discussion_r537880859 .
I would suggest to move this section into a `Maybe` section.

Reply:
I don't think supporting multiple Layers of Logic is a `Maybe` as we cannot provide a framework that doesn't allow for expressing certain logic. The two options for implementing multiple layers of logic, i.e. **Composition of Handlers** and **Composition of Property Wrappers** are marked as `proposal`.